src/lib.rs

```diff
@@ -11,7 +11,115 @@
 #![doc(html_root_url = "https://docs.rs/jsonapi/")]
 
 //! This is documentation for the `jsonapi` crate.
-//! The crate is meant to be used for serializing, deserializing and validating JSON-API requests and responses.
+//! The crate is meant to be used for serializing, deserializing and validating
+//! [JSON:API] requests and responses.
+//!
+//! [JSON:API]: https://jsonapi.org/
+//! [serde]: https://serde.rs
+//! [JsonApiDocument]: api/struct.JsonApiDocument.html
+//! [Resource]: api/struct.Resource.html
+//! [jsonapi_model]: macro.jsonapi_model.html
+//!
+//! ## Examples
+//!
+//! ### Basic Usage with Macro
+//!
+//! Using the [`jsonapi_model!`][jsonapi_model] macro a struct can be converted
+//! into a [`JsonApiDocument`][JsonApiDocument] or [`Resource`][Resource]. It is
+//! required that the struct have an `id` property whose type is `String`. The
+//! second argument in the [`jsonapi_model!`][jsonapi_model] marco defines the
+//! `type` member as required by the [JSON:API] specification
+//!
+//! ```rust
```

Comment:
Wanted to call this out - some of the contents of these examples is duplicative of the code that you already have defined in the tests. My primary motivation was to surface some examples so that people don't have to dig through the source code to see how this gets used.

That being said - this is pretty much taken directly from those tests. To prevent further duplication - do you think it is worth removing those tests from the `tests` directly and relying on documentation tests? These get run when `cargo test` is executed - so we maintain the current level of test coverage. Open to suggestions!

Reply:
> do you think it is worth removing those tests from the tests directly and relying on documentation tests?

Yes, absolutely. Removing duplication and moving relevant examples to doc tests is a great idea.

Personally I read docs for examples first and then start digging into tests if I can't find what I'm looking for